src/sage/numerical/backends/matrix_backend.pyx

```diff
@@ -0,0 +1,1170 @@
+    """
+    def __cinit__(self, maximization = True, base_ring=None, numpy_implementation = False, **kwds):
+        """
+        Cython constructor
+
+        EXAMPLES::
+
+            sage: from sage.numerical.backends.generic_backend import get_solver
+            sage: p = get_solver(solver = "Matrix")                 
+
+        """
+
+        #Sage Matrix and Vector instead of Python lists
+
+        self.prob_name = ''
+        self.obj_constant_term = 0
+        self.is_maximize = 1
+
+        #if base_ring is None:
+        #    from sage.rings.rational_field import QQ
+        #    base_ring = QQ
+        self._init_base_ring(base_ring=base_ring)
+
+        if numpy_implementation:
+            kwds = dict(implementation = "numpy")
+        else:
+            kwds = {}
+
+        self.objective_coefficients = matrix(self._base_ring, 1, [], **kwds)
+        self.Matrix = matrix(self._base_ring, [], **kwds)
+        self.row_lower_bound = matrix(self._base_ring, 1, [], **kwds)
+        self.row_lower_bound_indicator = []
+        self.row_upper_bound = matrix(self._base_ring, 1, [], **kwds)
+        self.row_upper_bound_indicator = []
+        self.col_lower_bound = matrix(self._base_ring, 1, [], **kwds)
+        self.col_lower_bound_indicator = []
+        self.col_upper_bound = matrix(self._base_ring, 1, [], **kwds)
+        self.col_upper_bound_indicator = []
+
+        self.row_name_var = []
+        self.col_name_var = []
+
+        self.is_continuous = []
+        self.is_binary = []
+        self.is_integer = []
+        if maximization:
+            self.set_sense(+1)
+        else:
+            self.set_sense(-1)
+
+        #self._base_ring = base_ring
+    
+    def _init_base_ring(self, base_ring=None):
+        if base_ring is None:
+            from sage.rings.rational_field import QQ
+            base_ring = QQ
+
+        self._base_ring = base_ring
+
+    cpdef base_ring(self):
+        """
+    The base ring
+
+    TESTS::
+
+            sage: from sage.numerical.backends.matrix_backend import MatrixBackend
+            sage: MatrixBackend(base_ring=QQ).base_ring()
+            Rational Field
+            sage: MatrixBackend(base_ring=AA).base_ring()
+            Algebraic Real Field
+            sage: MatrixBackend(base_ring=RDF).base_ring()
+            Real Double Field
+            sage: MatrixBackend(base_ring=RealField(100)).base_ring()
+            Real Field with 100 bits of precision
+
+        """
+        return self._base_ring
+    
+    cpdef __copy__(self):
+        """
+        Returns a copy of self.
+
+        EXAMPLES::
+
+            sage: from sage.numerical.backends.generic_backend import get_solver
+            sage: p = MixedIntegerLinearProgram(solver="Matrix")
+            sage: b = p.new_variable()
+            sage: p.add_constraint(b[1] + b[2] <= 6)
+            sage: p.set_objective(b[1] + b[2])
+            sage: cp = copy(p.get_backend())
+            sage: cp.solve()
+            0
```

Comment:
This doctest makes no sense because the `solve` method is not implemented!
Find a better way to verify in the doctests that a correct copy has been made.